packages/core/src/state-management/execution-queue.ts

```diff
@@ -44,15 +51,15 @@ export class ExecutionQueue {
         return this.tasks.add(docId.toString(), async () => {
           const doc = await this.get(docId);
```

Comment:
Should the semaphore be held around the call to `.get()`?  `get` might have to do real i/o after all

Reply:
True. My bad.

Reply:
Reduced responsibility of executionQ to just concurrency control. Can not make it accommodate for `Repository#get` and `Repository#load`.